wd_helper.au3

```diff
@@ -636,6 +644,202 @@ Func _WD_FrameLeave($sSession)
+	EndIf
+
+	#EndRegion - post processing
+
+	$sMessage = ($sMessage And $_WD_DEBUG = $_WD_DEBUG_Full) ? (' Information: ' & $sMessage) : ("")
+	Return SetError(__WD_Error($sFuncName, $iErr, $sParameters & $sMessage), $iExt, $vResult)
+EndFunc   ;==>_WD_FrameList
+
+; #INTERNAL_USE_ONLY# ===========================================================================================================
+; Name ..........: __WD_FrameList_Internal
+; Description ...: function that is used internally in _WD_FrameList, even recursively when nested frames are available
+; Syntax ........: __WD_FrameList_Internal(Const $sSession, Const $sLevel, $sFrameAttributes, Const $bReturnHTML)
+; Parameters ....: $sSession            - Session ID from _WD_CreateSession
+;                  $sLevel              - frame location level ... path
+;                  $sFrameAttributes    - all <iframe ....> atributes
+;                  $bReturnHTML         - On true function will return frame document HTML source
+; Return values .: Success - array or string
+;                  Failure - "" (empty string) and sets @error to one of the following values:
+;                  - $_WD_ERROR_Exception
+;                  - $_WD_ERROR_InvalidExpression
+; Author ........: mLipok
+; Modified ......: Danp2
+; Remarks .......:
+; Related .......: _WD_FrameList, _WD_GetFrameCount, _WD_FrameEnter, _WD_FrameLeave
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func __WD_FrameList_Internal(Const $sSession, Const $sLevel, $sFrameAttributes, Const $bReturnHTML)
+	Local Const $sFuncName = "__WD_FrameList_Internal"
+	Local Const $sParameters = 'Parameters:    Level=' & $sLevel & '   ReturnHTML=' & $bReturnHTML ; intentionally $sFrameAttributes is not listed here to not put too many data into the log
+	Local $iErr = $_WD_ERROR_Success
+	Local $vResult = '', $s_URL = '', $s_HTML = '', $sMessage = ''
+
+	_WD_FrameEnter($sSession, $sLevel)
+	$iErr = @error
+
+	If @error Then
+		$sMessage = 'Error occured on "' & $sLevel & '" level when trying to entering frame'
+	Else
+		Local $sCurrentBody_ElementID = _WD_ExecuteScript($sSession, "return window.document.body;", Default, Default, $_WD_JSON_Element)
+		$iErr = @error
+		If @error Then
+			$sMessage = 'Error occured on "' & $sLevel & '" level when checking "document.body" ElementID'
+		Else
+			$s_URL = _WD_ExecuteScript($sSession, "return window.location.href", Default, Default, $_WD_JSON_Value)
+			$iErr = @error
+			If @error Then
+				$sMessage = 'Error occured on "' & $sLevel & '" level when checking URL'
+			Else
+				If $bReturnHTML Then
+					$s_HTML = _WD_GetSource($sSession)
+					$iErr = @error
+					If @error Then
+						$s_HTML = ''
+						$sMessage = 'Error occured on "' & $sLevel & '" level when getting HTML Source'
+					Else
+						$s_HTML = StringToBinary($s_HTML, $SB_UTF8)
```

Comment:
Should we use `StringToBinary($s_HTML, $SB_UTF8)` or other for example: `$SB_ANSI` or even `__WD_Base64Encode` and `__WD_Base64Decode`

?

Reply:
Yes, something should be done here because it doesn't make sense to return the HTML as a binary string. Did you mean BinaryToString?

Reply:
Please check: 
[BinaryToString usage and HTML CleanUp](https://github.com/Danp2/au3WebDriver/pull/362/commits/a65e8c960b1ff8bf5c4a47d5094da76d8ca29dae)

Reply:
> Should we use `StringToBinary($s_HTML, $SB_UTF8)` or other for example: `$SB_ANSI`

I still wonder how character encoding on web pages follows and whether we can encode source HTML to BINARY with UTF8 or is it better to use ANSI ?

Reply:
It is a complex issue that could be avoided if you followed through on your recent statement 

> After rethinking I think that currently _WD_FrameList is even more complex then it should be with $sFilter and $bReturnHTML feature.

Reply:
I'm not likely to use this function, so I may not be the best judge of its functionality. I'll leave it to you to decide on the future of those features. If you decide to leave them in, then you should look at using `$_WD_BFORMAT` to control the binary formatting.

Reply:
Please take a look on: [Removal of $sFilter and $bReturnHTML](https://github.com/Danp2/au3WebDriver/pull/362/commits/83d6f4c5f7a04a505ff14aa98521233cb711820c)

Reply:
you can check it with modified wd_demo.au3: [DemoFrames _WD_FrameList and _WD_FrameEnter+path](https://github.com/Danp2/au3WebDriver/pull/362/commits/4073ade1b5581167ccd37de2f92d0585ca89b22b)